src/components/OrderBookWidget.tsx

```diff
@@ -224,103 +223,257 @@ function _printOrderBook(pricePoints: PricePointDetails[], baseToken: TokenDetai
+            const spread = calcSpread(maxBid, minAsk)
+            // How much will we zoom, based on the spread
+            const zoomInterval = calcZoomInterval(spread)
+            // Starting X value
+            const lowerZoomX = calcLowerZoomX(minBid, maxBid, minAsk, zoomInterval)
+            // Ending X value
+            const upperZoomX = calcUpperZoomX(minAsk, maxAsk, maxBid, zoomInterval)
+            // Ending Y value (Y always starts on 0)
+            const upperZoomY = calcUpperZoomY(bids, asks, lowerZoomX, upperZoomX)
+
+            // Calculate, given `upperZoomY`, where in % terms it fits
+            const endY = calcPercentageOfAxis(upperZoomY, 0, maxY)
+            // Same as above, for `lowerZoomX` and `upperZoomX`
+            startX = calcPercentageOfAxis(lowerZoomX, minX, range)
+            endX = calcPercentageOfAxis(upperZoomX, minX, range)
+
+            logDebug(`bids[${minBid.toFixed(10)}...${maxBid.toFixed(10)}]`)
+            logDebug(`asks[${minAsk.toFixed(10)}...${maxAsk.toFixed(10)}]`)
+            logDebug(`range: ${range}; minX: ${minX}`)
+            logDebug(`spread: ${spread}`)
+            logDebug(`zoomInterval: ${zoomInterval}`)
+            logDebug(`lowerZoomX: ${lowerZoomX}; upperZoomX: ${upperZoomX}`)
+            logDebug(`startX %: ${startX}; endX %: ${endX}`)
+            logDebug(`upperZoomY: ${upperZoomY}; maxY: ${maxY}`)
+            logDebug(`endY %: ${endY}`)
+
+            // Setting Y axis initial zoom
+            yAxis.end = endY
+          } else if (bids.length > 0) {
+            // There are no asks. Zoom on the right side of the graph
+            // TODO: magic number, move to const
+            xAxis.start = 0.95
+            xAxis.end = 1
+            // TODO: adjust yAxis
+          } else if (asks.length > 0) {
+            // There are no bids. Zoom on the left side of the graph
+            xAxis.start = 0
+            // TODO: magic number, move to const
+            xAxis.end = 0.5
+            // TODO: adjust yAxis
+          }
+
+          // Setting X axis initial zoom
+          xAxis.start = startX
+          xAxis.end = endX
+
+          _printOrderBook(pricePoints, baseToken, quoteToken)
+
+          return pricePoints
+        } catch (error) {
+          console.error('Error processing data', error)
+          return []
+        }
+      })
+
+      // Reload data from data source re-using same chart
+      chart.dataSource.load()
+
+      // Setting up tooltips based on currently loaded tokens
+      const [bidSeries, askSeries] = chart.series.values
+      bidSeries.tooltipText = `[bold]${market}[/]\nBid Price: [bold]{priceFormatted}[/] ${quoteTokenLabel}\nVolume: [bold]{totalVolumeFormatted}[/] ${baseTokenLabel}`
+      askSeries.tooltipText = `[bold]${market}[/]\nAsk Price: [bold]{priceFormatted}[/] ${quoteTokenLabel}\nVolume: [bold]{totalVolumeFormatted}[/] ${baseTokenLabel}`
+    }
+  }, [baseToken, chart, hops, networkId, quoteToken])
```

Comment:
Would also be great to reset zoom settings when `[baseToken, hops, networkId, quoteToken]` change

Reply:
But it does. Every time there's a change, it re-subscribes to the data event which calculates the zoom.
Not working like that for you?

Reply:
![ezgif-4-47dc9dbc405f](https://user-images.githubusercontent.com/5121491/88635064-e0f98000-d0bf-11ea-8735-e9c1728348e8.gif)
I it zoom setting not being reset or min/max recalculated each time to fit better?

Reply:
Nice gif, need to learn how to do that :)

Anyway, not sure yet what's wrong, it looks alright. When you change tokens the zoom is adjusted for the new data.

> ...  min/max recalculated each time to fit better?
Yes. Every time there's new data (token pair changes) the zoom is recalculated.